hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -715,17 +706,15 @@ def _prepare_empty_analytic_line(self):
         }
 
     def add_line(self):
+        self.ensure_one()
         if self.add_line_project_id:
-            values = self._prepare_empty_analytic_line()
-            name_line = self._get_line_name(
-                **self._get_new_line_name_values()
-            )
-            name_list = list(set(self.line_ids.mapped('value_y')))
-            if name_list:
+            task_list = self.line_ids.mapped('task_id')
```

Comment:
Grouping just by task is not correct, uniqueness of timesheet entry already depends on project and task, and other modules extend it

Reply:
Potentially, we could use tuple of IDs to compare

Reply:
This method is just for adding new lines. The grouping and the uniqueness check is done somewhere else. I only simplified the code, the actual behavior shouldn't be changed.